recipes/imagepy/meta.yaml

```diff
@@ -0,0 +1,46 @@
+{% set name = "imagepy" %}
+{% set version = "0.16" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "511556789365fffa4827f771ca246d10bbb7fd7a07240d6a29c39b3e44e83e1e" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.tar.gz'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ hash_value }}
+
+build:
+  number: 0
```

Comment:
Is this a pure Python package? If so, would take a look at using [`noarch: python`]( https://conda-forge.org/docs/meta.html#building-noarch-packages ). This would build one universal Python package that users could install on Python 2/3 on all platforms.

Reply:
yes, pure python package, (with numba, pure?)
@jakirkham I think it is the matter of wxpython, here: https://wxpython.org/pages/downloads/

"Because of the differences between Linux distributions (mainly different versions of the core libraries installed by default, but also platform architecture and etc.) it is not possible to host binary wheel files for Linux on PyPI..."

wxpython.org say pypi not work on linux, I do not know if conda con solve it? and is there any wx app on conda? I want to have a look!

Reply:
@yxdragon conda-forge has a `wxpython` package: https://github.com/conda-forge/wxpython-feedstock. It should work fine.

Reply:
Run time compilation via Numba still counts as pure Python. We are only concerned with compilation that occurs at build time and it sounds like `imagepy` doesn't do that. So `imagepy` counts as pure Python.

Reply:
@dougalsutherland 
that is my CI build log, libgtk-x11-2.0, so I think wx does not work.

from wx.core import *
  File "/home/conda/staged-recipes/build_artifacts/imagepy_1521691888499/_t_env/lib/python2.7/site-packages/wx/core.py", line 12, in <module>
    from ._core import *
ImportError: libgtk-x11-2.0.so.0: cannot open shared object file: No such file or directory

Reply:
Ack. Sorry about that.

So we are not as cleanly separated from the OS as we would like. In some cases, particularly GUI packages, we rely on system backed things like Gtk and X11 to be provided for us. Ideally we would package things like X11 (we have done a lot of work in that direction) and Gtk, which we are working on in PR ( https://github.com/conda-forge/staged-recipes/pull/3920 ). In cases where we rely on system packages, we notate this in a `yum_requirements.txt` file in the recipe like [this one for `wxpython`]( https://github.com/conda-forge/wxpython-feedstock/blob/30214f843b44442c2f3db660e1f117e5ce18689a/recipe/yum_requirements.txt ).

Would just copy [`wxpython`'s `yum_requirements.txt` file]( https://github.com/conda-forge/wxpython-feedstock/blob/30214f843b44442c2f3db660e1f117e5ce18689a/recipe/yum_requirements.txt ) and place it in your recipe here. This should install the needed Gtk and X11 libraries behind the scenes. In practice users will typically already have X11 and Gtk as they will be running on a Linux Desktop as opposed to a Docker image (which is where we are building this). In the future, we plan to do a better job of packaging these dependencies outright.

If you have any questions about this, please feel free to ask. :)

Reply:
Went ahead and added `yum_requirements.txt`, which should fix this. Hope that is ok.